addons/mail/static/src/web/messaging_service_patch.js

```diff
@@ -41,6 +42,26 @@ patch(Messaging.prototype, "mail/web", {
         }
         this._super(data);
     },
+    _handleNotificationRecordInsert(notif) {
+        this._super(notif);
+        if (notif.payload.Thread) {
+            const data = notif.payload.Thread;
+            const thread = this.store.threads[createLocalId(data.model, data.id)];
+            if (data.serverFoldState && data.serverFoldState !== thread.state) {
+                thread.state = data.serverFoldState;
+                if (thread.state === "closed") {
+                    const chatWindow = this.store.chatWindows.find(
+                        (chatWindow) => toRaw(chatWindow.thread) === toRaw(thread)
```

Comment:
I'd rather comparing `chatWindow.threadLocalId` which would not require the `toRaw` and would be more accurate than comparing objects.